backend/testfiles/execution/stdlib/httpclient.dark

```diff
@@ -178,271 +202,402 @@ module Disallowed =
+    )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://[0:0:0:0:0:0:0:0]"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Could not connect ([::]:80) Could not connect"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.ConnectionError
+      "Could not connect ([::]:80) Could not connect"
+  )
 
-  PACKAGE.Darklang.Stdlib.HttpClient.request "get" "localhost" [] Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Bad URL: Invalid URI"
+  PACKAGE.Darklang.Stdlib.HttpClient.request "get" "localhost" [] Builtin.Bytes.empty = PACKAGE
+    .Darklang
+    .Stdlib
+    .Result
+    .Result
+    .Error(
+      PACKAGE.Darklang.Stdlib.HttpClient.RequestError.BadUrl(
+        PACKAGE.Darklang.Stdlib.HttpClient.BadUrlDetails.InvalidUri
+      )
+    )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://localhost"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Bad URL: Invalid host"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.BadUrl(
+      PACKAGE.Darklang.Stdlib.HttpClient.BadUrlDetails.InvalidHost
+    )
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://127.0.0.1"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Bad URL: Invalid host"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.BadUrl(
+      PACKAGE.Darklang.Stdlib.HttpClient.BadUrlDetails.InvalidHost
+    )
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://[::1]"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Could not connect ([::1]:80) Could not connect"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.ConnectionError
+      "Could not connect ([::1]:80) Could not connect"
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://[0:0:0:0:0:0:0:1]"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Could not connect ([::1]:80) Could not connect"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.ConnectionError
+      "Could not connect ([::1]:80) Could not connect"
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://[0000:0000:0000:0000:0000:0000:0000:0001]"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Could not connect ([::1]:80) Could not connect"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.ConnectionError
+      "Could not connect ([::1]:80) Could not connect"
```

Comment:
Where does this text come from? (I'm curious if we can get rid of the redundant `Could not connect`

Reply:
Deep within dotnet - as I recall we have no access to it except in an Exception.Message.